extendr-macros/src/wrappers.rs

```diff
@@ -91,11 +91,27 @@ pub(crate) fn make_function_wrappers(
         quote! { #rust_name }
     };
 
+    // arguments for the wrapper with type being `SEXP`
     let formal_args = inputs
         .iter()
         .map(|input| translate_formal(input, self_ty))
         .collect::<syn::Result<Punctuated<FnArg, Token![,]>>>()?;
 
+    // extract the names of the arguments only (`mut` are ignored in `formal_args` already)
+    let sexp_args = formal_args
+        .clone()
+        .into_iter()
+        .map(|x| match x {
+            // the wrapper doesn't use `self` arguments
+            FnArg::Receiver(_) => unreachable!(),
```

Comment:
Is this 'never happens' or 'should not happen'? Maybe we can `panic!()` with a more informative message?

Reply:
I believe this is never happens, and will never happen. In the definition of `formal_args`, the `self` is replaced with a variable name, as `extern "C"` functions cannot take `Self`, and never will. Then, we merely extract the produced argument names, to have all the `SEXP`-typed variables in the wrapper available to us. 

But `unreachable` is just a panic, so if you want to suggest a message, then we can do that.